### ISSUE TITLE
JobGenerator can check the size of the queue and abort submission by threshold

### ISSUE THREAD

**Issue by niall-byrne** [CancerCollaboratory/sandbox#71]
From my experiences with working with Cecile, and operating Seqware workflows in multiple environments I have the following suggestions to improve the Queue system.

My initial reaction was that Queue was the wrong choice for a backbone, I am a fan of simple webservices- this is purely based off my experience supporting different environments during my career.

I could not articulate clearly why I thought this was a better solution, until now.

When operating the Queue system, the operator has to manage 3 states.

1) Ini files on the Launcher
2) Ini files as they travel through the queue
3) Ini files running on machines

Step 2, is a problem.  If you send multiple jobs into step 2, they sit in a state that is not monitored, is not reflected in GNOS or in ELASTICSEARCH, or on a RUNNING NODE, and there is a significant time delay now between steps 1 and 3.  

This introduces further problems, I can not upgrade workflows without draining the queue and removing this state.  I can not EASILY determine what stage of the process an ini file is at.  Expecting the operator to manage the Queue via the web interface seems unreasonable to me.

The solution I would propose is to leverage rabbitmq as a transport.  Take advantage of the fact that hosts can join the queue dynamically, and limit the uptake into the queue to a single ini file at a time.

If the Generator only ever reads one ini file at a time, it becomes much easier to manage state.
Even better, if you have a worker node inform the queue that it is ready for a job, the worker can trigger the intake process, and you never have a job waiting in the queue, and you never have to use this intermediary state.

Long story short, please don't put lots of jobs into the queue at the same time.  And it would be really cool if the worker could trigger the intake process.

**Comment by denis-yuen** [comment]
Hi,
As proposed, the scope is too broad for GitHub. 
I'm limiting this to the non-architectural change proposed which is to give the user of the system an easy way to determine how large the current queue is and to abort job submission when above a certain threshold. 

Architecturally, the current system is similar to [SGE](http://gridscheduler.sourceforge.net/) in that you can submit a large number of jobs to the queue. We can implement tools like qdel/qalter to delete jobs from the queue without major changes, but more drastic changes will need a broader discussion and more thought than what should be done on GitHub. 

Note: 
The current system does not actually cause the problems that you've defined
1) You do not need to drain the queue to upgrade a workflow. When changing the worker image, you just have to make sure that its compatible with the versions of the workflow in the queue. i.e. make sure your worker has both the new and old version of the workflow
2) It is easy to determine what state a job is in, you can either use our reporting tools or query postgres directly. We track whether an ini file has been submitted and is in the queue or is running/failed.

**Comment by niall-byrne** [comment]
That's fine we can discuss the architecture in another forum.
I feel like the system's methods for handling these concerns are suboptimal.
